src/main/java/org/kitteh/irc/client/library/feature/sts/STSPropertiesStorageManager.java

```diff
@@ -0,0 +1,199 @@
+        } catch (IOException e) {
+            throw new KittehSTSException(e.getMessage(), e);
+        }
+    }
+
+
+    /**
+     * @param hostname the hostname (as sent in the SNI by the client)
+     * @param duration the length (in seconds) until the expiry of this stored policy
+     * @param policy the STS policy instance, including all data sent from the server
+     */
+    @Override
+    public void addEntry(@Nonnull String hostname, long duration, @Nonnull STSPolicy policy) {
+        Sanity.nullCheck(hostname, "A valid hostname must be provided for this entry.");
+        Sanity.nullCheck(policy, "A valid policy must be provided to be inserted.");
+        this.properties.setProperty(hostname, getExpiryFromDuration(duration) + "; " + this.reserializeData(policy));
+        this.saveData();
+    }
+
+    /**
+     * Saves data using the provided Writer.
+     */
+    private void saveData() {
+        try {
+            final BufferedWriter bufferedWriter = Files.newBufferedWriter(this.filePath, StandardCharsets.UTF_8, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+            this.properties.store(bufferedWriter, "This file contains all the gathered STS policies.");
+            bufferedWriter.close();
+        } catch (IOException e) {
+            throw new KittehSTSException(e.getMessage());
+        }
+    }
+
+    /**
+     * Gets an STS policy from the store, looking it up via hostname.
+     *
+     * @param hostname the hostname (as sent in the SNI by the client)
+     * @return all data sent by the server in the CAP LS "sts" value when we connected securely
+     */
+    @Override
+    @Nonnull
+    public Optional<STSPolicy> getEntry(@Nonnull String hostname) {
+        Sanity.nullCheck(hostname, "A valid hostname must be provided for this entry.");
+
+        this.pruneEntries();
+        if (!this.hasEntry(hostname)) {
+            return Optional.empty();
+        }
+
+        String value = this.properties.getProperty(hostname);
+        String[] components = value.split("; ");
+        String data = components[1];
+        return Optional.of(STSUtil.getSTSPolicyFromString(",", data));
+    }
+
+    /**
+     * Checks if a policy has been stored for the hostname.
+     *
+     * @param hostname the hostname to check
+     * @return whether the entry exists in the store
+     */
+    @Override
+    public boolean hasEntry(@Nonnull String hostname) {
+        Sanity.nullCheck(hostname, "A valid hostname must be provided for this entry.");
+
+        this.pruneEntries();
+        return this.properties.containsKey(hostname);
+    }
+
+    /**
+     * Cleans up the outdated entries from the properties instance.
+     */
+    private void pruneEntries() {
+        Set<String> stagedRemovals = new HashSet<>();
+
+        for (String hostname : this.properties.stringPropertyNames()) {
+            String value = this.properties.getProperty(hostname);
+            String[] components = value.split("; ");
```

Comment:
> Is it clearly specified that there must be exactly one space (no more, no less)?

No, because it's an implementation detail specific to this persistence engine. The implementation is correct, users shouldn't care about the specifics.